### PR TITLE
add brightness_contrast function

### PR DESCRIPTION
**Pull request by aquatiko**
A handy function to visualize your fits images by changing brightness and contrast in your Jupyter notebooks.
Currently works for GrayScale images.

- [X] Add `brightness_contrast` function
- [X] Update Docs

- [x] Tweak to accept range from user (brightness and contrast)

Results in IJulia:
Screenshot from 2019-03-10 10-38-31

### CODE REVIEW

File: src/AstroImages.jl [hunk 1]
```diff
@@ -2,9 +2,9 @@ __precompile__()
 
 module AstroImages
 
-using FITSIO, FileIO, Images
+using FITSIO, FileIO, Images, Interact
```
Comment (aquatiko) [inline src/AstroImages.jl]: Also, I think Interact needs to be added to the dependencies and update CI for this to go green.

**Comment by aquatiko** [comment]
https://docs.opencv.org/3.4/d3/dc1/tutorial_basic_linear_transform.html
Reference for changing brightness and contrast using operation
`g(x) = a*f(x) + b`
where `a` controls contrast and `b` brightness

**Comment by aquatiko** [comment]
@giordano I have added the required changes but I don't quite understand the feature going to `showanime.jl` step.

File: REQUIRE [hunk 2]
```diff
@@ -3,3 +3,4 @@ FITSIO 0.12
 FileIO
 Images
 RecipesBase
+Interact
```
Comment (giordano) [inline REQUIRE]: I think that now also `ImageMagick` is necessary.

File: src/AstroImages.jl [hunk 3]
```diff
@@ -76,4 +76,23 @@ Base.convert(::Type{Matrix{C}}, img::AstroImage{T,C}) where {T,C<:Color} = rende
...
+"""
+function visualize(img::AstroImage{T,C}; brightness_range = (0, 255), contrast_range = (1,1000), threshold_range = (1, 255)) where {T,C}
+    @manipulate for brightness in brightness_range[1] : brightness_range[2], contrast in contrast_range[1] : contrast_range[2], threshold in threshold_range[1] : threshold_range[2]
+        tmp = img.data/255
+     @. tmp = (tmp * contrast) + brightness/255
+        
+        for ind in eachindex(tmp)
+            tmp[ind] = tmp[ind] >= threshold/255 ? tmp[ind] : 0 
+        end
+        colorview(Gray, tmp)             # Currently works for GrayScale images only
```
Comment (giordano) [inline src/AstroImages.jl]: I'd suggest using the same colour `C` as the parameter of the function, in place of hard-coded Gray.  Also, note that you can do
```julia
C.(tmp)
```
and if you put everything together with my previous comments you can find that the whole body of the `for` loop is a single line, without defining any temporary array

**Comment by giordano** [comment]
Please, also add a test for this function.  A `@testset` on its own is fine.

In general, it is very important to test every feature you introduce, or bug you fix.

**Comment by aquatiko** [comment]
@giordano the function `visualize` just shows the slider interface in IJulia. What should one asset for the tests?
Also, can you give some pointers on steps for `showmime.jl`

**Comment by giordano** [comment]
Don't worry about using it in showmime, it's trickier than I expected, I need to think more about it. I'll also the test.

**Comment by aquatiko** [comment]
@giordano would this require any more work before getting merged?

**Comment by giordano** [comment]
> would this require any more work before getting merged?

Not from you, thanks!

I'm waiting for an answer to this question: https://discourse.julialang.org/t/base-show-method-for-showing-a-widget-in-jupyter/22099  I initially thought that simply replacing the `render` function with the new function in
https://github.com/JuliaAstro/AstroImages.jl/blob/282b59e70ec522061f5bc39710c89f3df6adbe50/src/showmime.jl#L2-L3
would have been sufficient.  It turned out that this isn't the case, we probably need to use another MIME, but I don't know which one.

**Comment by giordano** [comment]
The trick is to use `text/html`: https://discourse.julialang.org/t/base-show-method-for-showing-a-widget-in-jupyter/22099/4  Would you like to have the honour to do the change? :slightly_smiling_face:

**Comment by aquatiko** [comment]
Yeah sure!! It would help me learn about the Jupyter interface we would be doing later

File: src/showmime.jl [hunk 4]
```diff
@@ -1,3 +1,22 @@
...
+"""
+function brightness_contrast(img::AstroImage{T,C}; brightness_range = 0:255,
+                             contrast_range = 1:1000) where {T,C}
+    @manipulate for brightness  in brightness_range, contrast in contrast_range
+        _brightness_contrast(C, img.data, brightness, contrast)
+    end
+end
+
 # This is used in Jupyter notebooks
 Base.show(io::IO, mime::MIME"image/png", img::AstroImage; kwargs...) =
```
Comment (giordano) [inline src/showmime.jl]: You can remove this method

Comment (aquatiko) [reply src/showmime.jl]: write `@. color(matrix / 255 * T(contrast) + T(brightness) / 255)` inside `brightness_contrast` instead?

Comment (giordano) [reply src/showmime.jl]: I'm not sure what you mean, I meant that you can remove the ` Base.show(io::IO, mime::MIME"image/png", img::AstroImage; kwargs...)` method, which this comment refers to.

**Comment by aquatiko** [comment]
@giordano Would this require further work before merging?

**Comment by aquatiko** [comment]
@giordano I think this needs a CI re-run.

**Pull request merged**